### PR TITLE
math: Implement math.nextafter

### PR DESCRIPTION
**Pull request by corona10**
https://docs.python.org/3.9/library/math.html#math.nextafter is introduced yesterday :)

### CODE REVIEW

**Comment by corona10** [comment]
@youknowone Can you please take a look?

File: tests/snippets/stdlib_math.py [hunk 1]
```diff
@@ -240,6 +240,15 @@ def assertAllNotClose(examples, *args, **kwargs):
 assert math.factorial(20) == 2432902008176640000
 assert_raises(ValueError, lambda: math.factorial(-1))
 
+assert math.nextafter(4503599627370496.0, -INF) == 4503599627370495.5
```
Comment (youknowone) [inline tests/snippets/stdlib_math.py]: if this is 3.9 feature, CPython in CI will not run this test. Adding version check will work

Comment (corona10) [reply tests/snippets/stdlib_math.py]: Oh, thanks, this is 3.9 feature, I will add it

**Comment by corona10** [comment]
@youknowon I 've updated the code only running on RustPython.

**Comment by windelbouwman** [comment]
Nice one! I think we should follow CPython while we are still not on par with cpython. Allthough this is a python 3.9 thing, I think we could still add it.

File: tests/snippets/stdlib_itertools.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import itertools
+import platform
```
Comment (youknowone) [inline tests/snippets/stdlib_itertools.py]: This line looks like a mistake

File: vm/src/stdlib/math.rs [hunk 3]
```diff
@@ -17,8 +17,15 @@ use crate::obj::objtype;
 use crate::pyobject::{PyObjectRef, PyResult, TypeProtocol};
 use crate::vm::VirtualMachine;
 
+use libc::c_double;
+
 use std::cmp::Ordering;
 
+#[link_name = "cmath"]
+extern "C" {
+    pub fn nextafter(x: c_double, y: c_double) -> c_double;
```
Comment (youknowone) [inline vm/src/stdlib/math.rs]: is this function need to be a `pub`?

File: tests/snippets/stdlib_itertools.py (hunk 2 above)
Comment (corona10) [reply tests/snippets/stdlib_itertools.py]: Thanks :)

File: vm/src/stdlib/math.rs (hunk 3 above)
Comment (corona10) [reply vm/src/stdlib/math.rs]: Not yet all

**Comment by corona10** [comment]
@youknowone Thanks for the review, Updated!

**Comment by corona10** [comment]
@youknowone @windelbouwman 
Finally, all CIs are passed.
- Change the unit test to be attribute-based. 
- raise NotImplementedError for WASM.

**Comment by youknowone** [comment]
Not sure `NotImplementedError` fit for this use, but basically it looks good.

**Pull request merged**